### PR TITLE
feature: cicads-619 linear sentence chunker

### PR DESCRIPTION
**Pull request by neil-stephen-mcgonigle**
## Summary
This PR introduces a new deterministic, sentence-aware chunking strategy for document ingestion, called the LineSentenceChunker. It also refactors the chunking architecture to support both the new line-based and the existing layout-based chunkers, improves configuration management, and updates the test suite for clarity and coverage.

### Key Changes
New Line-by-Line Sentence Chunker

- Implements LineSentenceChunker, which processes Textract LINE blocks directly, prioritizing sentence boundaries and vertical proximity.
- Ensures chunk indices start at zero for each page (per-page uniqueness contract).
- Adds configuration for word-based chunking and vertical gap detection.

###  Refactored Chunking Architecture

- Introduces a ChunkStrategy abstract base class with a clear contract for chunk index semantics.
- Adds a factory (get_chunk_strategy) to select between layout-based and line-based chunkers via config or environment variable.
- Cleans up and removes legacy/unused chunking config and code.
- Configuration Improvements

- All chunker settings are now available via .env and validated in config.py.
- Adds new settings for line-based chunker (min/max words, vertical gap ratio).
- Improves .env_template documentation and removes Python syntax errors.

### Test Suite Overhaul

- Updates and expands tests for both chunkers, including edge cases and contract validation.
- Removes or updates misleading comments and dead code.
- Ensures all tests expect per-page chunk indices and correct chunking behavior.

### Other Improvements

- Improves logging and error handling throughout the chunking pipeline.
- Updates shell scripts and local dev setup for better S3 key handling.
- Refactors method names for clarity (create_chunk replaces from_textractor_layout).
- 
### Motivation
- Addressed issues with the previous layout-based chunker, which sometimes produced unreliable or unintuitive chunks.
- Needed a deterministic, sentence-aware chunker for better downstream processing and search.
- Improved configuration and testability for both chunking strategies.
- Ensured robust, maintainable, and well-documented code for future enhancements.

### How to Use
- Select the chunking strategy via the DOCUMENT_CHUNKING_STRATEGY setting (linear-sentence-splitter or layout).
- Configure chunker parameters in .env as needed.
- Run the test suite to verify all chunking logic and edge cases.
- ```bash run_locally_with_dot_env.sh``` 

### Checklist
 - New chunker and factory pattern implemented
 - All settings and config validated
 - Tests updated and passing
 - Documentation and comments reviewed
 - Dead code and misleading comments removed

### CODE REVIEW

**Review by duncanabraham — COMMENTED** [review]
No issues found in the code. I've pulled the code and run it locally, all good. My only observation is there's low test coverage on these 3 files: 

```
src/ingestion_pipeline/chunking/strategies/layout/types/table/schemas.py                   28      7    75%   27, 36, 45, 54, 63, 72, 81
src/ingestion_pipeline/chunking/strategies/line_sentence/chunker.py                       121     27    78%   136-142, 150-166, 238, 254, 262-268
src/ingestion_pipeline/chunking/verbose_page_debug_logger.py                               34     23    32%   27-42, 60-72
```

File: src/ingestion_pipeline/chunking/strategies/line_sentence/chunker.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+            current_word_count += line_word_count
+            prev_line_bottom = line_bbox.y + line_bbox.height
+
+            # --- Gap break: close chunk including the gap-triggering line ---
+            if should_break_on_gap:
+                chunk_index = self._emit_chunk(chunks, current_lines, page_number, metadata, chunk_index, gap_reason)
+                current_lines = []
+                current_word_count = 0
+                i += 1
+                continue
```
[the remark this answers is not in the document]
Comment (neil-stephen-mcgonigle) [reply src/ingestion_pipeline/chunking/strategies/line_sentence/chunker.py]: Changed as per the recommendation, the chunk is emitted before adding the new line.

**Review by duncanabraham — DISMISSED** [review]
LGTM 👍

**Comment by neil-stephen-mcgonigle** [comment]
Merged as part  of https://github.com/ministryofjustice/cica-review-case-documents-airflow/pull/200
